smac/utils/validate.py

```diff
@@ -37,15 +39,34 @@ def _unbound_tae_starter(
     ----------
     tae: ExecuteTARun
         tae to be used
+    runhistory: RunHistory
+        runhistory to save
+    run_info: RunInfo
+        Config to be launched
     *args, **kwargs: various
         arguments to the tae
 
     Returns
     -------
-    tae_results: tuple
+    tae_results: RunValue
         return from tae.start
     """
-    return tae.start(*args, **kwargs)
+    result = execute_ta_run_wrapper(tae, run_info, *args, **kwargs)
+    tae.stats.ta_runs += 1
+    tae.stats.ta_time_used += float(result.time)
+    if runhistory:
```

Comment:
It can be optional, please look at https://github.com/automl/SMAC3/blob/d5b93811aa74e0513e580576ebb0fbf857e18f1a/test/test_utils/test_validate.py#L80

I will agree however that static checking should have highlighted, that it should be typing.Optional and I will add this. 


I am not at all familiar with the validation, but looking at the unit testing, the owner expect it to be None in certain cases.